### ISSUE TITLE
socks-Proxy not working - Long vs Integer for socksVersion param

### ISSUE THREAD

**Issue by Sadrak** [SeleniumHQ/selenium#8774]
## 🐛 Bug Report

We try to use Selenium with socks-proxies from Perl via Webdrive 3 but can't initialize the session:
`Could not create new session: java.lang.Long cannot be cast to java.lang.Integer at test.pl line 26.`

Here is the logoutput from the hub:
```
13:22:35.415 INFO [ActiveSessionFactory.apply] - Capabilities are: {
  "acceptSslCerts": true,
  "browserName": "chrome",
  "goog:chromeOptions": {
  },
  "javascriptEnabled": true,
  "proxy": {
    "proxyType": "manual",
    "socksProxy": "selenium_proxy_1:1080",
    "socksVersion": 5
  },
  "version": ""
}
[1602508955.421][SEVERE]: bind() failed: Cannot assign requested address (99)
```

When we "fix" the socksVersion-Parsing in `java/client/src/org/openqa/selenium/Proxy.java` everything work as expected.
When we switch to http-Proxy everything work as expected.

Perhaps this is a combination of perl + java and typecasting, but Selenium should handle userinput better.

Our quickfix:
```
--- a/java/client/src/org/openqa/selenium/Proxy.java
+++ b/java/client/src/org/openqa/selenium/Proxy.java
@@ -91,7 +91,13 @@ public Proxy(Map<String, ?> raw) {
       setSocksProxy((String) raw.get("socksProxy"));
     }
     if (raw.containsKey("socksVersion") && raw.get("socksVersion") != null) {
-      setSocksVersion(((Number) raw.get("socksVersion")).intValue());
+      //setSocksVersion(((Number) raw.get("socksVersion")).intValue());
+      try {
+        setSocksVersion((Integer) raw.get("socksVersion"));
+      } catch (ClassCastException exc) {
+        Long l = (Long) raw.get("socksVersion");
+        setSocksVersion(l.intValue());
+      }
     }
     if (raw.containsKey("socksUsername") && raw.get("socksUsername") != null) {
       setSocksUsername((String) raw.get("socksUsername"));
```

## To Reproduce

Start a Selenium hub with a chrome/firefox browser and try to use socksVersion.

## Expected behavior

Using Selenium over a Proxy.

## Environment

OS: Debian
Browser: chrome/firefox
Browser version: unknown
Browser Driver version: chromedriver: 2.35, geckodriver: 0.19.1
Language Bindings version: Perl

**Comment by barancev** [comment]
What version of Selenium Server do you use?

**Comment by Sadrak** [comment]
docker hub: selenium/hub:latest
Created: 2020-10-10T00:55:39.940728233Z
Selenium server version: 3.141.59, revision: e82be7d358

**Comment by Sadrak** [comment]
is an duplicated #6917

Can we please get an backport to 3.*?

**Comment by diemol** [comment]
Thanks for pointing that it is a duplicate, @Sadrak.
However, we do not port fixes back to previous releases. We suggest starting to use the new Grid 4, even if it is on Alpha state, the docker images are working well already.

**Comment by Sadrak** [comment]
@diemol the error ist very old and hit many installations. Also we tried to jump von Grid 4, but can't get everything up and running. So we are stuck with 3 with an old bug.

We have now an quickfix (using gost as additional chained proxy for "http proxy" to "socks proxy").

**Comment by diemol** [comment]
If you need help moving to Grid 4, just jump to our Slack/IRC channel and let us know what problems you are having. The links can be seen here https://www.selenium.dev/support/

**Comment by xtianus** [comment]
@Sadrak I managed to build a patched version of 3.141.59 and the "Long vs Integer" problem is gone, but I still can't connect to a Socks5 proxy with password using Chrome.
You're welcome to my binaries: https://drive.google.com/drive/folders/1Z_7_ibx1eLJLrtiBERNrXelWaH-wW6y0?usp=sharing
